zcash_primitives/src/transaction/components/transparent/pczt.rs

```diff
@@ -0,0 +1,214 @@
+    ///   cannot produce signatures for this sighash type must not provide a signature.
+    /// - Spend Finalizers must fail to finalize inputs which have signatures that do not
+    ///   match this sighash type.
+    pub(crate) sighash_type: SighashType,
+
+    /// A map from a pubkey to the BIP 32 derivation path at which its corresponding
+    /// spending key can be found.
+    ///
+    /// - The pubkeys should appear in `script_pubkey` or `redeem_script`.
+    /// - Each entry is set by an Updater.
+    /// - Individual entries may be required by a Signer.
+    pub(crate) bip32_derivation: BTreeMap<[u8; 33], Bip32Derivation>,
+
+    /// Mappings of the form `key = RIPEMD160(value)`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) ripemd160_preimages: BTreeMap<[u8; 20], Vec<u8>>,
+
+    /// Mappings of the form `key = SHA256(value)`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) sha256_preimages: BTreeMap<[u8; 32], Vec<u8>>,
+
+    /// Mappings of the form `key = RIPEMD160(SHA256(value))`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) hash160_preimages: BTreeMap<[u8; 20], Vec<u8>>,
+
+    /// Mappings of the form `key = SHA256(SHA256(value))`.
+    ///
+    /// - These may be used by the Signer to inspect parts of `script_pubkey` or
+    ///   `redeem_script`.
+    pub(crate) hash256_preimages: BTreeMap<[u8; 32], Vec<u8>>,
+
+    /// Proprietary fields related to the transparent coin being spent.
+    pub(crate) proprietary: BTreeMap<String, Vec<u8>>,
+}
+
+/// Information about a transparent output within a transaction.
+///
+/// This struct is for representing transparent outputs in a partially-created
+/// transaction. If you have a fully-created transaction, use
+/// [the regular `TxOut` struct].
+///
+/// [the regular `TxOut` struct]: super::TxOut
+#[derive(Debug, Getters)]
+#[getset(get = "pub")]
+pub struct Output {
+    /// The value of the output.
+    pub(crate) value: Zatoshis,
+
+    /// The script constraining how spending of this output must be authorized.
+    pub(crate) script_pubkey: Script,
+
+    /// The script required to spend this output, if it is P2SH.
+    ///
+    /// Set to `None` if this is a P2PKH output, or a P2SH with an unknown redeem script.
+    pub(crate) redeem_script: Option<Script>,
+
+    /// A map from a pubkey to the BIP 32 derivation path at which its corresponding
+    /// spending key can be found.
+    ///
+    /// - The pubkeys should appear in `script_pubkey` or `redeem_script`.
+    /// - Each entry is set by an Updater.
+    /// - Individual entries may be required by a Signer.
+    pub(crate) bip32_derivation: BTreeMap<[u8; 33], Bip32Derivation>,
```

Comment:
Same question here, what is this used for on the output side?

Reply:
I suspect the intention (from BIP 174) was to enable Signers to verify that they can indeed produce signatures for change outputs (and therefore verify that the change is going to an address they can subsequently spend from).